packages/main/src/Input.ts

```diff
@@ -1011,7 +1011,7 @@ class Input extends UI5Element implements SuggestionComponent, IFormInputElement
 		const inputDomRef = this.getInputDOMRefSync();
 		const emptyValueFiredOnNumberInput = this.value && this.isTypeNumber && !inputDomRef!.value;
 		const eventType: string = (e as InputEvent).inputType
-			|| (e.detail as InputEventDetail).inputType
+			|| (e.detail as InputEventDetail)?.inputType
```

Comment:
This is only to fix the runtime error. As I sent you more info offline, we can hide the native clear icon completely (with standard CSS) and suggest using ours.